akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -528,14 +561,33 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
               state.registry.allUniqueAddressesInState(setup.selfUniqueAddress)
             val notInCluster = allAddressesInState.diff(state.registry.nodes)
 
-            if (notInCluster.isEmpty) Behaviors.same
-            else {
+            if (notInCluster.nonEmpty) {
               if (ctx.log.isDebugEnabled)
                 ctx.log.debug2(
                   "ClusterReceptionist [{}] - Leader node cleanup tick, removed nodes: [{}]",
                   cluster.selfAddress,
                   notInCluster.mkString(","))
-              nodesRemoved(notInCluster)
+              // FIXME config of delay
+              import scala.concurrent.duration._
+              ctx.scheduleOnce(5.seconds, ctx.self, CleanupRemoved(notInCluster))
```

Comment:
This is something I'm uncertain about. Can we solve this in a better way?
The problem is that we might have entries from ddata but not the membership information yet, so it looks like stale entries that should be removed.

Reply:
I don't quite get this scenario, the periodic tick would notice nodes in cluster that were removed, but here instead we defer it wit ha delay so that a ddata remove can arrive inbetween? I don't see the problem with the service being removed "twice".

Reply:
The problem is when the tick happens before we have seen the node as joining, but we have already got the ddata update. Then it looks like we have an entry in the ddata registry that is supposed to be removed, because the node isn't there.